### PR TITLE
Switch to TravisCi and Coveralls

### PR DESCRIPTION
**Pull request by deweydbb**

### CODE REVIEW

File: bin/git/hooks/pre-commit-golang [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        LINTERRORS=true
+    fi
+
+    # vet everything except test files
+    if ! [[ "${FILE}" == *"_test.go"* ]]; then
+        go vet "${FILE}"
+        if [[ $? != 0 ]]; then
+            VETERRORS=true
+        fi
+    fi
```
Comment (dkoston) [inline bin/git/hooks/pre-commit-golang]: around here, we should add in `git add ${FILE}` I believe to restage these after the changes

File: bin/git/hooks/pre-commit-golang [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+gofmt -w ./..
+
+LINTERRORS=false
+VETERRORS=false
+
+go vet ./...
+if [[ $? != 0 ]]; then
+    VETERRORS=true
+fi
+
```
Comment (deweydbb) [inline bin/git/hooks/pre-commit-golang]: There was a problem with vetting a single file at a time. In limiter.go we define the limiter struct as having the RequestStatus and RateLimitConfig structs, which are defined in another file. When vetting a single file at a time, I do not believe it looks at the whole package and so it says that RequestStatus and RateLimitConfig structs are undefined. 

The code above vets the whole package but I think it also vets the test files, which were excluded when vetting on e file at a time.

File: bin/git/hooks/pre-commit-golang [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+gofmt -w ./..
+
+LINTERRORS=false
+VETERRORS=false
+
+go vet ./...
+if [[ $? != 0 ]]; then
+    VETERRORS=true
+fi
```
Comment (deweydbb) [inline bin/git/hooks/pre-commit-golang]: There was a problem with vetting a single file at a time. In limiter.go we define the limiter struct as having the RequestStatus and RateLimitConfig structs, which are defined in another file. When vetting a single file at a time, I do not believe it looks at the whole package and so it says that RequestStatus and RateLimitConfig structs are undefined.

The code above vets the whole package but I think it also vets the test files, which were excluded when vetting one file at a time.

**Comment by dkoston** [comment]
Thanks, I’m cool with vetting test files. I had excluded them due to issues with vetting single files at a time so vetting whole package seems to be the way to go

**Comment by dkoston** [comment]
It would be nice also to add a script which will fail if anything from “go vet”, goincludes, golint, or gofmt failed or changed a file. That way we could run it in Travis CI and fail the build if those tools weren’t run on the commit

**Comment by dkoston** [comment]
Looks good to me. Thanks!

**Pull request merged**